src/python/pants/engine/target_test.py

```diff
@@ -866,3 +875,98 @@ def hydrate(sources_cls: Type[Sources], sources: Iterable[str]) -> HydratedSourc
             "f2.txt",
             "f3.txt",
         )
+
+
+# -----------------------------------------------------------------------------------------------
+# Test Codegen
+# -----------------------------------------------------------------------------------------------
+
+
+class AvroSources(CodegenSources):
+    pass
+
+
+class AvroLibrary(Target):
+    alias = "avro_library"
+    core_fields = (AvroSources,)
+
+
+class GenerateFortranFromAvroRequest(GenerateSourcesRequest):
+    input = AvroSources
+    output = FortranSources
```

Comment:
NB: this is technically a lie. We are not returning `FortranSources`. Rather, we are returning `GeneratedSources` with a Snapshot of Fortran files.

I originally added a new type called `CodegenTargetLanguage` so that this would be `output = Fortran`. I got rid of it because that is still a lie, and it led to too much boilerplate.